lib/stateView/document/query/findNotIndexedOrderByFields.js

```diff
@@ -0,0 +1,164 @@
+/**
+ * @typedef {[string, string]} indexStructure
+ */
+
+/**
+ * Check if conditions satisfy indices:
+ * index fields are in the same order, that sorting fields
+ * and have the same direction
+ *
+ * @param {indexStructure[]} indexedFields
+ * @param {indexStructure[]} sortingFields
+ * @returns {boolean}
+ */
+function areSortingFieldsIndexedInRightOrder(indexedFields, sortingFields) {
+  if (sortingFields.length < indexedFields.length) {
+    return false;
+  }
+
+  return indexedFields
+    .every(([field, direction], i) => {
+      const [conditionField, conditionDirection] = sortingFields[i];
+
+      return field === conditionField && direction === conditionDirection;
+    });
+}
+
+/**
+ * check if we have all indexed fields in where condition
+ * @param {indexStructure[]} compoundIndexBefore
+ * @param {[string]} whereConditions
+ * @returns {boolean}
+ */
+function areIndexedFieldsPresentInWhereCondition(compoundIndexBefore, whereConditions) {
+  return compoundIndexBefore
+    .every(([field]) => whereConditions
+      .some(conditionField => field === conditionField));
+}
+
+/**
+ * Validate query to sort by only indexed fields
+ *
+ * @typedef findNotIndexedOrderByFields
+ * @param {{field, direction}[][]} dataContractIndexFields
+ * @param {indexStructure[]} sortingFields
+ * @param {[string, string, string][]} [whereConditions]
+ * @returns {Array}
+ */
+function findNotIndexedOrderByFields(
+  dataContractIndexFields,
+  sortingFields,
+  whereConditions = [],
+) {
+  // flat where conditions and set sort directions
+  const compiledWhere = whereConditions.reduce(
```

Comment:
Let's just walk through whereCondition in `areIndexedFieldsPresentInWhereCondition` and remove this part?